core/modules/rewrite.cc

```diff
@@ -107,7 +107,7 @@ inline void Rewrite::DoRewrite(bess::PacketBatch *batch) {
 
   next_turn_ = start + cnt;
   if (next_turn_ >= bess::PacketBatch::kMaxBurst) {
-    next_turn_ -= bess::PacketBatch::kMaxBurst;
+    next_turn_ = next_turn_ % num_templates_;
```

Comment:
If I remember correctly it added about 1-2 cycles per "packet", although I expect the performance impact will highly depend on what modules are used downstream (i.e., whether those modules use any integer division/modulo instructions).